Parsers/ASimAuthentication/Parsers/ASimAuthenticationGoogleWorkspace.yaml

```diff
@@ -0,0 +1,153 @@
+      EventResult: string,
+      DvcAction: string
+  )
+          [
+      "login_success", "Logon", "Success", "Allowed",
+      "login_failure", "Logon", "Failure", "Blocked",
+      "login_challenge", "Logon", "", "",
+      "login_verification", "Logon", "", "",
+      "risky_sensitive_action_blocked", "Logon", "Failure", "Blocked",
+      "riskay_sensitive_action_allowed", "Logon", "Success", "Allowed",
+      "logout", "Logoff", "Success", "Allowed",
+      "suspicious_login", "Logon", "Failure", "Blocked",
+      "suspicious_login_less_secure_app", "Logon", "Failure", "Blocked",
+      "suspicious_programmatic_login", "Logon", "Failure", "Blocked",
+      "user_signed_out_due_to_suspicious_session_cookie", "Logoff", "Success", "Allowed"
+  ];
+      let ThreatEventTypes = dynamic(['suspicious_login', 'suspicious_login_less_secure_app', 'suspicious_programmatic_login', 'user_signed_out_due_to_suspicious_session_cookie']);
+      let SupportedEventNames = EventFieldsLookup
+          | project EventOriginalSubType;
+      union isfuzzy=true  GoogleWorkspaceSchema, GWorkspace_ReportsAPI_login_CL
+      | where not(disabled)
+      | where event_name_s in (SupportedEventNames)
+      | lookup EventFieldsLookup on $left.event_name_s == $right.EventOriginalSubType
+      | project-rename
+          TargetUsername = actor_email_s,
+          TargetUserId = actor_profileId_s,
+          SrcIpAddr = IPAddress,
+          LogonMethod = login_challenge_method_s,
+          EventOriginalType = event_type_s,
+          EventOriginalUid = id_uniqueQualifier_s
+      | extend
+          TargetUsername = iif(event_name_s in (ThreatEventTypes), affected_email_address_s, TargetUsername),
+          TargetUsernameType = _ASIM_GetUsernameType(TargetUsername),
+          TargetUserIdType = iif(isnotempty(TargetUserId), "GWorkspaceProfileID", ""),
+          EventSeverity = iif(event_name_s in (ThreatEventTypes), "High", "Informational")
+      | extend 
+          AdditionalFields = bag_pack(
+                        "Is_Suspicious",
+                        is_suspicious_b,
+                        "Is_Second_Factor_b",
+                        is_second_factor_b,
+                        "Logon_Type",
+                        login_type_s,
+                        "Sensitive_Action_Name",
+                        sensitive_action_name_s
+                    ),
+          EventResult = case(
+                    event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "passed",
+                    "Success",
+                    event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "incorrect_answer_entered",
+                    "Failure",
+                    EventResult
+                ),
+          EventResultDetails = iif(event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "incorrect_answer_entered", "MFA not satisfied", ""),
```

Comment:
is Blank accepted for enumeration field, do you mean NA ?

Reply:
@anki-narravula EventResultDetails is not a mandatory field. can be kept as blank. Here, I want to set this value only when event_name_s  is 'login_challenge' or 'login_verification'